### ISSUE TITLE
Using semantic role names instead of colors in default stylesheet

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor#1664]
The default stylesheet should provide semantic role names rather than specific color names. The color name roles should be removed. (https://github.com/asciidoctor/asciidoctor/blob/v1.5.4/data/stylesheets/asciidoctor-default.css#L323-L354)

Here's a list of the suggested names (taking a page from Bootstrap and Foundation):

* primary
* secondary
* success
* info
* warning
* danger (or maybe important)
* caution
* highlight
* alert

...as well as the corresponding bg- roles. See [Contextual Colors](http://getbootstrap.com/css/#helper-classes-colors) in the Bootstrap documentation for reference and ideas.

Users can, of course, define their own role names using docinfo.

**Comment by Mogztter** [comment]
+1
But I think we should also keep the color name because some colors (lime,
maroon, fuschia...) don't have a "logical" role associated.

And we should also use a nicer palette like http://clrs.cc/
IMO default CSS colors are ugly :)

**Comment by mojavelinux** [comment]
> IMO default CSS colors are ugly :)

I agree. We actually don't use the default colors as it is now. We use some color shift logic to make them a tiny less ugly.

>  I think we should also keep the color name because some colors (lime,maroon, fuschia...)

I don't agree because I don't think color names belong in content. Every color should have some purpose associated with it. In other words, the color is a concern of the design/theme/styles, not the content.

Of course, if you really want to use color names, you can define all the colors in the rainbow as roles using docinfo. But we should not be encouraging this by having color names available by default. I think this is one way where AsciiDoc Python broke its own rule for encouraging the use of semantic content.

**Comment by mojavelinux** [comment]
I should add that semantic colors also pave the way for themes sensitive to or dedicated for color blindness. We can focus on the behavior of the color and not the actual appearance. Because what is blue to one person might be gold to another.

**Comment by tedbergeron** [comment]
I've created an Asciidoctor page to demonstrate the semantic names ( http://tedbergeron.github.io/AsciidoctorSemanticColors/ ) 

I have added icons to suggest the correlation between the semantic names and the admonition blocks. 

- Info aka NOTE. 
- Danger aka IMPORTANT (as Dan suggested)

I'm not endorsing the color palette that I came up with. It's just an example of trying to fit in with the default asciidoctor.css theme. Consider it guiding inspiration. 

I am not a fan of the secondary color I picked, it just happened to be one of the contrasting colors found on the palette generator. I was trying to make caution = yellow, warning = orange, danger = red and ran out of colors.

IMO the background color should still allow black text to be legible, so it would highlight black text.

I'm not sure how to go about coming up with a contrasting palette for color blind readers.

**Comment by Mogztter** [comment]
> I don't agree because I don't think color names belong in content. Every color should have some purpose associated with it. In other words, the color is a concern of the design/theme/styles, not the content.

Fair enough.

> Because what is blue to one person might be gold to another.

You are not talking about THE :dress: ?

> I've created an Asciidoctor page to demonstrate the semantic colors

Wow this is nice, I love it!

> I'm not sure how to go about coming up with a contrasting palette for color blind readers.

You can use this bookmarklet: https://github.com/Altreus/colourblind (not sure if this is realistic though)
As you can see there are different kind of color blindness:
 * protanomaly
 * protanopia
 * deuteranomaly
 * deuteranopia
 * tritanomaly
 * tritanopia
 * achromatomaly
 * achromatopsia

I think the important part is to have a good contrast.
Overall your proposal is quite good :smile:

**Comment by mojavelinux** [comment]
> You are not talking about THE ￼:dress: ?

:wink:

**Comment by danyill** [comment]
Today I thought I'd file an issue saying I wanted to see a full :rainbow: of colours to use. Then I found this fine issue and I think @mojavelinux provides two very fine reasons why this should continue to be discouraged.

* Retaining semantic meaning and separating presentation from content
* Designing for alternative or changing stylesheets (branding) and for accessibility

However for anyone coming from a WYSIWYG word processor this is something of an adjustment. Perhaps when the Asciidoctor docs are Antora-ized we could potentially add this as an FAQ question.

I really like the palette of @tedbergeron and at some stage it would be neat to see a refresh of the default stylesheet colours.

**Comment by sdavids** [comment]
Related: https://gitlab.com/antora/antora-ui-default/-/issues/221
